CMakeLists.txt

```diff
@@ -186,6 +186,14 @@ if(MSVC)
 	set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -MP -W4 ${MSVC_DISABLED_WARNINGS_STR}")
 endif()
 
+include(TestBigEndian)
+TEST_BIG_ENDIAN(IS_BIG_ENDIAN)
+if(IS_BIG_ENDIAN)
+	add_definitions(-DHAVE_BIG_ENDIAN)
+else()
+	add_definitions(-DHAVE_LITTLE_ENDIAN)
+endif()
```

Comment:
Suggesting to guard this with `if(MSVC)`, to avoid doing this check unnecessarily with gcc/clang toolchains.

Reply:
Right yes it makes sense. Thank you. It sounds better to separate it, so this is the fallback path. I have update the changes with it, if it looks better!